hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationBackupHandler.java

```diff
@@ -188,14 +218,21 @@ private Data getBackupOperationData(BackupAwareOperation backupAwareOp) {
         // if getServiceName() method is overridden to return the same name
         // then this will have no effect.
         backupOp.setServiceName(op.getServiceName());
-        return nodeEngine.getSerializationService().toData(backupOp);
+        return backupOp;
     }
 
-    private Backup newBackup(BackupAwareOperation backupAwareOp, Data backupOpData, long[] replicaVersions,
+    private Backup newBackup(BackupAwareOperation backupAwareOp, Object backupOp, long[] replicaVersions,
             int replicaIndex, boolean respondBack) {
-
         Operation op = (Operation) backupAwareOp;
-        Backup backup = new Backup(backupOpData, op.getCallerAddress(), replicaVersions, respondBack);
+        Backup backup;
+        if (backupOp instanceof Operation) {
+            backup = new Backup((Operation) backupOp, op.getCallerAddress(), replicaVersions, respondBack);
```

Comment:
In fact, I don't like having two fields for same purpose inside `Backup` and only one of them is used. 
Another solution I had though was using two `Backup` implementation as `Data` based and `Operation` based for backup operation but it seemed to me more complex than this one.